scripts/dynamic_free_pool.py

```diff
@@ -0,0 +1,183 @@
+                                #line = line.replace(old_project, new_project)
+				words = line.split()
+				words[1] = new_project
+				line = ' '.join(words) + "\n"
+                        lines.append(line)
+        with open(statusfile,'w') as status_file:
+                for line in lines:
+                        status_file.write(line)
+
+#Queries current status from HaaS for a node.
+def get_node_current_info(haas_url, haas_username, haas_password, node):
+	'''
+	Checks the current information for the node and returns a tuple consisting of (<project_name>,List of <{"network": network-name}>, NIC name)
+	'''
+	url = haas_url + "node/" + node
+	resp = requests.get(url, auth = (haas_username,haas_password))
+	node_info = resp.json()
+	node_project = node_info['project']
+	data = node_info['nics'][0]['networks']
+	node_nic = node_info['nics'][0]['label']
+	networks = [{"network":v} for k,v in data.items() if k.startswith('vlan')]
+	return (node_project, networks, node_nic)	
+
+#Detach networks from nodes.
+def detach_networks_from_node(haas_url, haas_username, haas_password, node, networks, node_nic):
+	'''Takes a node ,list of {network, "network-name"}- networks attached to it, nic on which network is attached and detaches all the networks
+	'''
+	url = haas_url + "node/" + node + "/nic/" + node_nic + "/detach_network"
+	for network in networks:
+		res = requests.post(url, json.dumps(network), auth = (haas_username,haas_password))
+		time.sleep(5)
+	return res	
+
+
+#Updates the file with project name as free and releases it back to free pool
+def release_from_project(haas_url, haas_username, haas_password, statusfile, node, project):	
+	lines = []
+        with open(statusfile,'r') as status_file:
+                for line in status_file:
+                        if node in line:
+                                line = node +" "+ "free_pool "+str(0) +"\n"
+                        lines.append(line)
+	with open(statusfile,'w') as status_file:
+                for line in lines:
+                        status_file.write(line)
+	time.sleep(5)
+	response = release_node_from_project(haas_url, haas_username, haas_password,\
+						node, project)
+	return response	
+		
+#Release a node from a project. 
+def release_node_from_project(haas_url, haas_username, haas_password, node, project):
+	haas_url = haas_url + 'project/' + project + '/detach_node'
+	body = {
+		"node" : node
+	       }
+	r = requests.post(haas_url, data = json.dumps(body), auth = (haas_username, haas_password))
+	return r
+	
+#Checks for free pool in the list and returns the list
+def get_free_node_list(haas_url, haas_user_name, haas_password):	
+	free_node_list =  urlparse.urljoin(haas_url,'free_nodes')
+	free_node_list = requests.get(free_node_list, auth = (haas_user_name, haas_password))
+	return free_node_list
+
+
+#For checking purpose
+if __name__ == "__main__":
+	try:
+		config = ConfigParser.SafeConfigParser()
+		config.read("/home/ravi/leasing.cfg")
```

Comment:
Actually this should be moved to /etc/, so that we can read from that location instead of having it in some user's home location. But as of now, since I am reading from my account, I did it that way. Also, we need to run it from haas or some other account.